src/visor/visor.go

```diff
@@ -380,21 +390,59 @@ func (vs *Visor) CreateAndExecuteBlock() (coin.SignedBlock, error) {
 	return sb, err
 }
 
+// CreateBlock creates a Block from specified set of transactions according to set of determinstic rules.
+func (vs *Visor) CreateBlock(txns coin.Transactions, when uint64) (coin.Block, error) {
```

Comment:
The name should match the function it calls, in this case `CreateBlockFromTxns`, and make sure the db.Update label matches too